File: scala/src/main/scala/day01/SonarSweep1.scala
package day01

import common.Common

object SonarSweep1 {

  def main(args: Array[String]): Unit = {
    val values = Common.readFile("/day01/input.txt").map { value => value.toInt }

    val increases = values.zip(values.drop(1))
      .count { case (u, v) => u < v }

    println(increases)
  }
}

File: kotlin/src/main/kotlin/day01/SonarSweep1.kt
package day01

import Common

fun main() {
    val values = Common.readInput("/day01/input.txt").map(String::toInt)
    println(SonarSweep1().countIncreases(values))
}

class SonarSweep1 {

    fun countIncreases(values: List<Int>): Int {
        var increases = 0

        for (i in 1 until values.size) {
            if (values[i] > values[i - 1]) {
                increases++
            }
        }

        return increases
    }
}
